lib/SIL/SILFunctionType.cpp

```diff
@@ -1546,7 +1546,8 @@ namespace {
       }
 
       auto type = tl.getLoweredType().getSwiftRValueType();
-      if (type->hasRetainablePointerRepresentation())
+      if (type->hasRetainablePointerRepresentation()
+          || (type->getSwiftNewtypeUnderlyingType() && !tl.isTrivial()))
```

Comment:
Are newtypes either trivial or a retainable pointer? Why doesn't hasRetainablePointerRepresentation() return true for them?